junit5/src/main/java/sunstone/core/AzureUtils.java

```diff
@@ -0,0 +1,100 @@
+package sunstone.core;
+
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.management.profile.AzureProfile;
+import com.azure.identity.ClientSecretCredentialBuilder;
+import com.azure.resourcemanager.AzureResourceManager;
+import com.azure.resourcemanager.appservice.models.WebApp;
+import com.azure.resourcemanager.compute.models.VirtualMachine;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.wildfly.extras.sunstone.api.impl.ObjectProperties;
+import org.wildfly.extras.sunstone.api.impl.ObjectType;
+
+import java.io.IOException;
+import java.net.SocketTimeoutException;
+import java.util.Set;
+
+public class AzureUtils {
+    private static ObjectProperties objectProperties = new ObjectProperties(ObjectType.JUNIT5, null);
+
+    static AzureResourceManager getResourceManager() {
+        return AzureResourceManager
+                .authenticate(getCredentials(), new AzureProfile(AzureEnvironment.AZURE))
+                .withSubscription(objectProperties.getProperty(JUnit5Config.Azure.SUBSCRIPTION_ID));
+    }
+
+    private static TokenCredential getCredentials() {
+        return new ClientSecretCredentialBuilder()
+                .tenantId(objectProperties.getProperty(JUnit5Config.Azure.TENANT_ID))
+                .clientId(objectProperties.getProperty(JUnit5Config.Azure.APPLICATION_ID))
+                .clientSecret(objectProperties.getProperty(JUnit5Config.Azure.PASSWORD))
+                .build();
+    }
+
+    static boolean propertiesForArmClientArePresent() {
+        return objectProperties.getProperty(JUnit5Config.Azure.SUBSCRIPTION_ID) != null
+                && objectProperties.getProperty(JUnit5Config.Azure.TENANT_ID) != null
+                && objectProperties.getProperty(JUnit5Config.Azure.APPLICATION_ID) != null
+                && objectProperties.getProperty(JUnit5Config.Azure.PASSWORD) != null;
+    }
+
+    static VirtualMachine findAzureVM(AzureResourceManager arm, String name, Set<String> resourceGroups) {
+        for (String rg : resourceGroups) {
+            VirtualMachine vm = arm.virtualMachines().getByResourceGroup(rg, name);
+            if (vm != null) {
+                return vm;
+            }
+        }
+        return null;
+    }
+
+    static WebApp findAzureWebApp(AzureResourceManager arm, String name, Set<String> resourceGroups) {
+        for (String rg : resourceGroups) {
+            WebApp app = arm.webApps().getByResourceGroup(rg, name);
+            if (app != null) {
+                return app;
+            }
+        }
+        return null;
+    }
+
+    static void waitForWebApp(WebApp app) throws InterruptedException, IOException {
+        OkHttpClient client = new OkHttpClient();
+        // 20 minutes in millis
+        long timeout = TimeoutUtil.adjust(1200000);
```

Comment:
Suggestion: Timeout could be configurable using property and `waitForWebApp()` could be overloaded by `timeout` parameter.

Reply:
It can be adjusted by the `timeout.factor`. By default it is `1`, but you can set it to `0.5` which means 10 minutes timeout.

Reply:
I'm ok with current solution. Above was suggestion we can evaluate in the future.